### ISSUE TITLE
The performance is bad

### ISSUE THREAD

**Issue by YuqingWang1029** [poodarchu/DETR.detectron2#4]
Thanks for your work. After I trained using you code for about 70 epochs, the AP is only 0.003. Is there any problem I should notice? Can you share your training log? Also, I found the checkpoint generated by this code is 317M, but the checkpoint from your provided link is 159M. Why are they different?

 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.003
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.011
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.005
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.005
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.006
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.033
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.057
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.080
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.013
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.069
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.161

**Comment by poodarchu** [comment]
I only keep model parameters to reduce checkpint's size, which config do you use?

**Comment by poodarchu** [comment]
![image](https://user-images.githubusercontent.com/11813362/85198008-baedfc80-b317-11ea-9896-0bb96422601f.png)
you can refer to this screenshot to check whether your training process is valid.

**Comment by poodarchu** [comment]
The most important metric to take care for is class_error, I guess in your model, class_error might be always 100.00.

**Comment by poodarchu** [comment]
this is an example training log of the 1st epoch. https://github.com/poodarchu/DETR.detectron2/files/4807719/log.txt

**Comment by YuqingWang1029** [comment]
@poodarchu Thanks for your quick reply. I used the config of detr.res50.coco.multiscale.150e.yaml  and found my loss value is totally different from yours. Also, there is no class_error value and other error values in my log. Could you download your own code and watch the log? 

06/20 20:23:40 d2.engine.train_loop]: Starting training from iteration 0
[06/20 20:24:05 d2.utils.events]:  eta: 4 days, 0:16:55  iter: 19  total_loss: 203.594  loss_ce: 2.638  loss_bbox: 4.624  loss_giou: 2.263  loss_ce_0: 2.408  loss_bbox_0: 4.783  loss_giou_0: 2.235  loss_ce_1: 2.397  loss_bbox_1: 4.757  loss_giou_1: 2.242  loss_ce_2: 2.542  loss_bbox_2: 4.644  loss_giou_2: 2.269  loss_ce_3: 2.535  loss_bbox_3: 4.534  loss_giou_3: 2.288  loss_ce_4: 2.553  loss_bbox_4: 4.441  loss_giou_4: 2.294  time: 0.3130  data_time: 0.9438  lr: 0.000100  max_mem: 3218M
[06/20 20:24:11 d2.utils.events]:  eta: 3 days, 21:41:41  iter: 39  total_loss: 185.240  loss_ce: 2.039  loss_bbox: 3.429  loss_giou: 2.711  loss_ce_0: 2.054  loss_bbox_0: 3.448  loss_giou_0: 2.669  loss_ce_1: 2.029  loss_bbox_1: 3.395  loss_giou_1: 2.720  loss_ce_2: 2.037  loss_bbox_2: 3.343  loss_giou_2: 2.772  loss_ce_3: 2.043  loss_bbox_3: 3.341  loss_giou_3: 2.803  loss_ce_4: 2.007  loss_bbox_4: 3.354  loss_giou_4: 2.801  time: 0.3083  data_time: 0.0070  lr: 0.000100  max_mem: 3463M
[06/20 20:24:17 d2.utils.events]:  eta: 3 days, 20:49:45  iter: 59  total_loss: 193.070  loss_ce: 2.044  loss_bbox: 3.281  loss_giou: 2.772  loss_ce_0: 2.056  loss_bbox_0: 3.288  loss_giou_0: 2.811  loss_ce_1: 2.042  loss_bbox_1: 3.266  loss_giou_1: 2.799  loss_ce_2: 2.042  loss_bbox_2: 3.238  loss_giou_2: 2.818  loss_ce_3: 2.010  loss_bbox_3: 3.216  loss_giou_3: 2.807  loss_ce_4: 2.015  loss_bbox_4: 3.232  loss_giou_4: 2.817  time: 0.3055  data_time: 0.0077  lr: 0.000100  max_mem: 3463M
[06/20 20:24:24 d2.utils.events]:  eta: 3 days, 19:59:24  iter: 79  total_loss: 196.116  loss_ce: 2.200  loss_bbox: 3.168  loss_giou: 2.738  loss_ce_0: 2.237  loss_bbox_0: 3.163  loss_giou_0: 2.787  loss_ce_1: 2.209  loss_bbox_1: 3.173  loss_giou_1: 2.756  loss_ce_2: 2.216  loss_bbox_2: 3.165  loss_giou_2: 2.746  loss_ce_3: 2.172  loss_bbox_3: 3.145  loss_giou_3: 2.740  loss_ce_4: 2.170  loss_bbox_4: 3.129  loss_giou_4: 2.749  time: 0.3039  data_time: 0.0074  lr: 0.000100  max_mem: 3463M
[06/20 20:24:29 d2.utils.events]:  eta: 3 days, 19:35:55  iter: 99  total_loss: 180.560  loss_ce: 1.893  loss_bbox: 3.185  loss_giou: 2.632  loss_ce_0: 1.941  loss_bbox_0: 3.178  loss_giou_0: 2.704  loss_ce_1: 1.916  loss_bbox_1: 3.172  loss_giou_1: 2.671  loss_ce_2: 1.893  loss_bbox_2: 3.150  loss_giou_2: 2.641  loss_ce_3: 1.888  loss_bbox_3: 3.142  loss_giou_3: 2.619  loss_ce_4: 1.867  loss_bbox_4: 3.140  loss_giou_4: 2.632  time: 0.3022  data_time: 0.0070  lr: 0.000100  max_mem: 3463M
[06/20 20:24:36 d2.utils.events]:  eta: 3 days, 19:29:12  iter: 119  total_loss: 188.533  loss_ce: 2.067  loss_bbox: 2.959  loss_giou: 2.582  loss_ce_0: 2.127  loss_bbox_0: 3.028  loss_giou_0: 2.601  loss_ce_1: 2.061  loss_bbox_1: 3.003  loss_giou_1: 2.589  loss_ce_2: 2.077  loss_bbox_2: 2.939  loss_giou_2: 2.575  loss_ce_3: 2.056  loss_bbox_3: 2.906  loss_giou_3: 2.617  loss_ce_4: 2.018  loss_bbox_4: 2.911  loss_giou_4: 2.603  time: 0.3017  data_time: 0.0067  lr: 0.000100  max_mem: 3463M
[06/20 20:24:42 d2.utils.events]:  eta: 3 days, 19:27:23  iter: 139  total_loss: 192.423  loss_ce: 2.051  loss_bbox: 2.985  loss_giou: 2.620  loss_ce_0: 2.064  loss_bbox_0: 3.007  loss_giou_0: 2.661  loss_ce_1: 2.040  loss_bbox_1: 3.008  loss_giou_1: 2.627  loss_ce_2: 2.034  loss_bbox_2: 2.939  loss_giou_2: 2.594  loss_ce_3: 2.086  loss_bbox_3: 2.936  loss_giou_3: 2.630  loss_ce_4: 2.064  loss_bbox_4: 2.966  loss_giou_4: 2.614  time: 0.3013  data_time: 0.0074  lr: 0.000100  max_mem: 3463M
[06/20 20:24:48 d2.utils.events]:  eta: 3 days, 19:12:15  iter: 159  total_loss: 177.000  loss_ce: 1.928  loss_bbox: 2.457  loss_giou: 2.377  loss_ce_0: 1.935  loss_bbox_0: 2.545  loss_giou_0: 2.432  loss_ce_1: 1.941  loss_bbox_1: 2.471  loss_giou_1: 2.356  loss_ce_2: 1.914  loss_bbox_2: 2.402  loss_giou_2: 2.310  loss_ce_3: 1.905  loss_bbox_3: 2.398  loss_giou_3: 2.347  loss_ce_4: 1.910  loss_bbox_4: 2.405  loss_giou_4: 2.355  time: 0.3008  data_time: 0.0070  lr: 0.000100  max_mem: 3463M

**Comment by Epiphqny** [comment]
> @poodarchu Thanks for your quick reply, in my log there is no class_error. And I use batchsize of 32 on 8 gpus. I will check what happens.
> 
> ^[[35mfc1000_b^[[0m
> 1952 ^[[35mfc1000_w^[[0m
> 1953 ^[[35mconv1_b^[[0m
> 1954 [06/18 16:51:25] d2.engine.train_loop INFO: Starting training from iteration 0
> 1955 [06/18 16:51:57] d2.utils.events INFO: eta: 3 days, 1:36:22 iter: 19 total_loss: 203.430 loss_ce: 2.673 loss_bbox: 4.864 loss_giou: 2.234 loss_ce_0: 2.486 loss_bbox_0: 4.669 loss_giou_0: 2.274 loss_ce_1: 2.444 loss_bbox_1 : 4.571 loss_giou_1: 2.301 loss_ce_2: 2.422 loss_bbox_2: 4.626 loss_giou_2: 2.274 loss_ce_3: 2.491 loss_bbox_3: 4.552 loss_giou_3: 2.298 loss_ce_4: 2.613 loss_bbox_4: 4.662 loss_giou_4: 2.258 time: 0.5788 data_time: 1.25 93 lr: 0.000100 max_mem: 4293M
> 1956 [06/18 16:52:08] d2.utils.events INFO: eta: 3 days, 1:33:53 iter: 39 total_loss: 188.178 loss_ce: 2.167 loss_bbox: 3.552 loss_giou: 2.701 loss_ce_0: 2.173 loss_bbox_0: 3.584 loss_giou_0: 2.761 loss_ce_1: 2.169 loss_bbox_1 : 3.502 loss_giou_1: 2.829 loss_ce_2: 2.186 loss_bbox_2: 3.513 loss_giou_2: 2.796 loss_ce_3: 2.128 loss_bbox_3: 3.469 loss_giou_3: 2.823 loss_ce_4: 2.146 loss_bbox_4: 3.481 loss_giou_4: 2.759 time: 0.5725 data_time: 0.20 33 lr: 0.000100 max_mem: 4293M
> 1957 [06/18 16:52:21] d2.utils.events INFO: eta: 3 days, 1:33:45 iter: 59 total_loss: 194.078 loss_ce: 2.178 loss_bbox: 3.393 loss_giou: 2.775 loss_ce_0: 2.181 loss_bbox_0: 3.371 loss_giou_0: 2.851 loss_ce_1: 2.184 loss_bbox_1 : 3.322 loss_giou_1: 2.870 loss_ce_2: 2.190 loss_bbox_2: 3.382 loss_giou_2: 2.808 loss_ce_3: 2.160 loss_bbox_3: 3.378 loss_giou_3: 2.803 loss_ce_4: 2.172 loss_bbox_4: 3.349 loss_giou_4: 2.792 time: 0.5577 data_time: 0.16 31 lr: 0.000100 max_mem: 4709M
> 1958 [06/18 16:52:32] d2.utils.events INFO: eta: 3 days, 1:51:33 iter: 79 total_loss: 188.344 loss_ce: 2.056 loss_bbox: 3.284 loss_giou: 2.664 loss_ce_0: 2.092 loss_bbox_0: 3.344 loss_giou_0: 2.724 loss_ce_1: 2.056 loss_bbox_1 : 3.286 loss_giou_1: 2.723 loss_ce_2: 2.046 loss_bbox_2: 3.301 loss_giou_2: 2.676 loss_ce_3: 2.023 loss_bbox_3: 3.255 loss_giou_3: 2.671 loss_ce_4: 2.045 loss_bbox_4: 3.251 loss_giou_4: 2.653 time: 0.5605 data_time: 0.22 97 lr: 0.000100 max_mem: 4709M
> 1959 [06/18 16:52:43] d2.utils.events INFO: eta: 3 days, 1:51:26 iter: 99 total_loss: 191.091 loss_ce: 2.124 loss_bbox: 3.084 loss_giou: 2.622 loss_ce_0: 2.143 loss_bbox_0: 3.207 loss_giou_0: 2.684 loss_ce_1: 2.109 loss_bbox_1 : 3.162 loss_giou_1: 2.649 loss_ce_2: 2.105 loss_bbox_2: 3.144 loss_giou_2: 2.605 loss_ce_3: 2.112 loss_bbox_3: 3.085 loss_giou_3: 2.613 loss_ce_4: 2.093 loss_bbox_4: 3.083 loss_giou_4: 2.585 time: 0.5570 data_time: 0.19 48 lr: 0.000100 max_mem: 4709M
> 1960 [06/18 16:52:57] d2.utils.events INFO: eta: 3 days, 1:47:35 iter: 119 total_loss: 185.125 loss_ce: 2.010 loss_bbox: 2.930 loss_giou: 2.676 loss_ce_0: 2.062 loss_bbox_0: 3.096 loss_giou_0: 2.727 loss_ce_1: 2.028 loss_bbox_ 1: 2.998 loss_giou_1: 2.691 loss_ce_2: 2.018 loss_bbox_2: 2.966 loss_giou_2: 2.660 loss_ce_3: 2.032 loss_bbox_3: 2.916 loss_giou_3: 2.653 loss_ce_4: 2.013 loss_bbox_4: 2.907 loss_giou_4: 2.634 time: 0.5586 data_time: 0.2 185 lr: 0.000100 max_mem: 4864M
> 1961 [06/18 16:53:08] d2.utils.events INFO: eta: 3 days, 1:47:28 iter: 139 total_loss: 185.771 loss_ce: 2.097 loss_bbox: 2.639 loss_giou: 2.513 loss_ce_0: 2.132 loss_bbox_0: 2.903 loss_giou_0: 2.633 loss_ce_1: 2.088 loss_bbox_ 1: 2.761 loss_giou_1: 2.533 loss_ce_2: 2.080 loss_bbox_2: 2.670 loss_giou_2: 2.487 loss_ce_3: 2.066 loss_bbox_3: 2.601 loss_giou_3: 2.479 loss_ce_4: 2.060 loss_bbox_4: 2.598 loss_giou_4: 2.457 time: 0.5578 data_time: 0.2 115 lr: 0.000100 max_mem: 4864M
> 1962 [06/18 16:53:19] d2.utils.events INFO: eta: 3 days, 1:47:21 iter: 159 total_loss: 181.855 loss_ce: 2.089 loss_bbox: 2.470 loss_giou: 2.335 loss_ce_0: 2.100 loss_bbox_0: 2.686 loss_giou_0: 2.457 loss_ce_1: 2.093 loss_bbox_ 1: 2.494 loss_giou_1: 2.319 loss_ce_2: 2.081 loss_bbox_2: 2.402 loss_giou_2: 2.270 loss_ce_3: 2.129 loss_bbox_3: 2.385 loss_giou_3: 2.284 loss_ce_4: 2.097 loss_bbox_4: 2.468 loss_giou_4: 2.269 time: 0.5550 data_time: 0.1 928 lr: 0.000100 max_mem: 4864M
> 1963 [06/18 16:53:31] d2.utils.events INFO: eta: 3 days, 1:43:01 iter: 179 total_loss: 177.928 loss_ce: 2.147 loss_bbox: 2.089 loss_giou: 2.077 loss_ce_0: 2.183 loss_bbox_0: 2.382 loss_giou_0: 2.316 loss_ce_1: 2.196 loss_bbox_ 1: 2.132 loss_giou_1: 2.077 loss_ce_2: 2.137 loss_bbox_2: 2.055 loss_giou_2: 2.040 loss_ce_3: 2.187 loss_bbox_3: 2.018 loss_giou_3: 1.999 loss_ce_4: 2.140 loss_bbox_4: 2.030 loss_giou_4: 2.015 time: 0.5526 data_time: 0.2 859 lr: 0.000100 max_mem: 4864M
> 1964 [06/18 16:53:41] d2.utils.events INFO: eta: 3 days, 1:42:04 iter: 199 total_loss: 174.741 loss_ce: 2.071 loss_bbox: 2.030 loss_giou: 1.975 loss_ce_0: 2.132 loss_bbox_0: 2.230 loss_giou_0: 2.070 loss_ce_1: 2.109 loss_bbox_ 1: 1.925 loss_giou_1: 1.925 loss_ce_2: 2.091 loss_bbox_2: 1.935 loss_giou_2: 1.950 loss_ce_3: 2.100 loss_bbox_3: 1.988 loss_giou_3: 1.960 loss_ce_4: 2.077 loss_bbox_4: 2.019 loss_giou_4: 1.938 time: 0.5465 data_time: 0.1 497 lr: 0.000100 max_mem: 4864M
> 1965 [06/18 16:53:52] d2.utils.events INFO: eta: 3 days, 1:44:13 iter: 219 total_loss: 173.192 loss_ce: 2.018 loss_bbox: 1.842 loss_giou: 1.908 loss_ce_0: 2.070 loss_bbox_0: 2.077 loss_giou_0: 1.992 loss_ce_1: 2.038 loss_bbox_ 1: 1.828 loss_giou_1: 1.815 loss_ce_2: 2.008 loss_bbox_2: 1.781 loss_giou_2: 1.828 loss_ce_3: 2.006 loss_bbox_3: 1.757 loss_giou_3: 1.850 loss_ce_4: 2.002 loss_bbox_4: 1.800 loss_giou_4: 1.835 time: 0.5438 data_time: 0.1 657 lr: 0.000100 max_mem: 4864M

**Comment by poodarchu** [comment]
I think you're using official detectron2.
You can remove the official detectron2, and rebuild my version.

Your problem is, you are actually using official detectron2 to run this repo.

**Comment by YuqingWang1029** [comment]
@poodarchu Thank you very much for your instruction, I rebuilt the code and the log is normal now.

**Comment by zzzzzz0407** [comment]
@Epiphqny What is the main difference btween offical detectron2 and this code? Different optimizer matters?

**Comment by poodarchu** [comment]
> @Epiphqny What is the main difference btween offical detectron2 and this code? Different optimizer matters?

1. Optimizer. Only AdamW can be used for DETR training.
2. LOGGING is also different
3. >= PyTorch 1.3 will be ok
